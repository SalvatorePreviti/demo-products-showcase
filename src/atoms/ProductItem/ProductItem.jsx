import React from 'react'
import PropTypes from 'prop-types'
import styles from './ProductItem.scss'

const headerStyle = styles.header
const headerActiveStyle = `${styles.header} ${styles.active}`

function ProductItem(props) {
  const { title, active, description } = props
  return (
    <div>
      <a
        href="#"
        className={active ? headerActiveStyle : headerStyle}
        onClick={event => {
          if (props.onSelect) {
            props.onSelect()
          }
          if (event) {
            event.preventDefault()
          }
        }}>
        {title}
      </a>
      {active ? <div className={styles.description}>{description}</div> : null}
    </div>
  )
}

ProductItem.propTypes = {
  title: PropTypes.string.isRequired,
  active: PropTypes.bool,
  description: PropTypes.string,
  onSelect: PropTypes.func
}

export default ProductItem
