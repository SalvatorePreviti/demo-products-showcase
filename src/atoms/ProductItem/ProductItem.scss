.header {
  display: block;
  text-decoration: none;

  &,
  &:visited,
  &:active,
  &:focus {
    color: #000;
  }

  &.active {
    font-weight: bold;
  }
}

.description {
  padding-bottom: 15px;
}
