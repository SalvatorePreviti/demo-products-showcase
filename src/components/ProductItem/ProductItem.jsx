import React from 'react'
import PropTypes from 'prop-types'
import styles from './ProductItem.scss'

const headerStyle = styles.header
const headerActiveStyle = styles.header + ' ' + styles.active
const contentStyle = styles.content

function ProductItem(props) {
  const { itemId, title, active, description } = props
  return (
    <div>
      <a
        href="#"
        className={active ? headerActiveStyle : headerStyle}
        onClick={() => {
          props.onSelect && props.onSelect(itemId)
        }}>
        {title}
      </a>
      {active ? <div className={contentStyle}>{description}</div> : null}
    </div>
  )
}

ProductItem.propTypes = {
  itemId: PropTypes.string,
  title: PropTypes.string.isRequired,
  active: PropTypes.bool,
  description: PropTypes.string,
  onSelect: PropTypes.func
}

export default ProductItem
