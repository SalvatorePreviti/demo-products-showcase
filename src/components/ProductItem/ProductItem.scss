.header {
  display: block;
  text-decoration: none;

  &,
  &:visited,
  &:active,
  &:focus {
    color: #000;
  }

  &.active {
    font-weight: bold;
  }
}

.content {
  padding-bottom: 15px;
}
