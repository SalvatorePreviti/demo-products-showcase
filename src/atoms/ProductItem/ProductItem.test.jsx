import React from 'react'
import { shallow } from 'enzyme'

import ProductItem from './ProductItem'

describe('ProductItem', () => {
  describe('not active', () => {
    let root

    beforeAll(() => {
      root = shallow(<ProductItem title="Item title" />)
    })

    it('snapshots', () => {
      expect(root).toMatchSnapshot()
    })

    it('renders header', () => {
      const header = root.find('a')
      expect(header.props()).toMatchObject({
        href: '#',
        className: 'header',
        children: 'Item title'
      })
    })

    it('does not have description visible', () => {
      expect(root.find('div.description')).toHaveLength(0)
    })
  })

  describe('active', () => {
    let root

    beforeAll(() => {
      root = shallow(<ProductItem title="Active item title" description="Item description" active={true} />)
    })

    it('snapshots', () => {
      expect(root).toMatchSnapshot()
    })

    it('renders header with active style', () => {
      const header = root.find('a')
      expect(header.props()).toMatchObject({
        href: '#',
        className: 'header active',
        children: 'Active item title'
      })
    })

    it('renders description', () => {
      const description = root.find('div.description')
      expect(description.props()).toMatchObject({
        className: 'description',
        children: 'Item description'
      })
    })
  })

  describe('when clicking', () => {
    it('invokes onSelect with the item id', () => {
      let selected
      const root = shallow(
        <ProductItem
          title="Item"
          onSelect={() => {
            selected = true
          }}
        />
      )

      root
        .find('a')
        .at(0)
        .simulate('click')

      expect(selected).toBe(true)
    })
  })
})
