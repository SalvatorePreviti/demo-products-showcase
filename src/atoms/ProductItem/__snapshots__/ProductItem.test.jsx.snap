// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem active snapshots 1`] = `
<div>
  <a
    className="header active"
    href="#"
    onClick={[Function]}
  >
    Active item title
  </a>
  <div
    className="description"
  >
    Item description
  </div>
</div>
`;

exports[`ProductItem not active snapshots 1`] = `
<div>
  <a
    className="header"
    href="#"
    onClick={[Function]}
  >
    Item title
  </a>
</div>
`;
